src/AutoRest.CSharp/LowLevel/Generation/LowLevelDataPlaneClientWriter.cs

```diff
@@ -317,12 +319,13 @@ private void WriteClientMethodBody(CodeWriter writer, LowLevelClientMethod clien
         private void WriteClientMethodDecleration(CodeWriter writer, RestClientMethod clientMethod, LowLevelOperationSchemaInfo operationSchemas, bool async)
         {
             var parameters = clientMethod.Parameters.Concat(new Parameter[] { RequestOptionsParameter });
+            var isConstantResponseBody = clientMethod.Responses.All(response => response.ResponseBody is ConstantResponseBody body) && clientMethod.ReturnType != null && clientMethod.ReturnType.FrameworkType == typeof(bool);
```

Comment:
Ok, so in what case `clientMethod.ReturnType != null && clientMethod.ReturnType.Equals(typeof(bool))` is `true`, but `clientMethod.Responses.All(response => response.ResponseBody is ConstantResponseBody body)` is `false`?

Reply:
For an example: https://github.com/Azure/autorest.csharp/blob/feature/v3/test/TestServerProjects/body-boolean/Generated/BoolClient.cs#L41-L71, in these cases return type will be true.

Reply:
Ok, so the `isConstantResponseBody` is `true` only when `request.HttpMethod == RequestMethod.Head && _context.Configuration.HeadAsBoolean` is true. Condition above is implicit.

I'd prefer to either add a "HeadAsBoolean" read-only property to the `RestClientMethod` or check HttpMethod here directly.

Reply:
Instead of adding the property, we can directly check `Configuration.HeadAsBoolean`